index.js

```diff
@@ -470,21 +494,45 @@ const createFocusTrap = function (elements, userOptions) {
         return this;
       }
 
-      updateTabbableNodes();
-
       state.active = true;
       state.paused = false;
       state.nodeFocusedBeforeActivation = doc.activeElement;
 
-      const onActivate =
-        activateOptions && activateOptions.onActivate
-          ? activateOptions.onActivate
-          : config.onActivate;
-      if (onActivate) {
-        onActivate();
+      const getOption = (optionName) => {
+        return activateOptions && activateOptions[optionName] !== undefined
+          ? activateOptions[optionName]
+          : config[optionName];
+      };
+
+      const onActivate = getOption('onActivate');
+      const onSuccessfulActivation = getOption('onSuccessfulActivation');
+      const checkCanActivate = getOption('checkCanActivate');
+
+      if (checkCanActivate) {
+        if (onActivate) {
+          onActivate();
+        }
+
+        state.containers.forEach((container) => {
+          delayFocusTrapActivation(container, checkCanActivate, () => {
+            updateTabbableNodes();
+            addListeners();
+            if (onSuccessfulActivation) {
+              onSuccessfulActivation();
+            }
```

Comment:
Since this is the same pattern to execute whether we delay or not, let's put this in a local function and call it in both places so we're always calling the same `finishActivation()` steps.

Reply:
I would love to but we need to be comfortable with the idea that it will potentially break some existing code bases since it swaps the call order of `updateTabbableNodes()` and `onActivate()` around.

Reply:
I managed to work around the issue by checking if `checkCanFocusTrap` exists.

If yes then it waits until that resolves to call `updateTabbableNodes()`

If not, then it calls it at the same place it did before to prevent breaking peoples code.